pkg/test/script.go

```diff
@@ -0,0 +1,101 @@
+package test
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+type File struct {
+	Name string
+	Data string
+}
+
+type Shell struct {
+	Name     string
+	Script   string
+	Input    []File
+	Expected []File
+}
+
+type ShellTest struct {
+	Shell
+	scriptFile *os.File
+	dir        string
+}
+
+func NewShellTest(s Shell) *ShellTest {
+	return &ShellTest{Shell: s}
+}
+
+// Setup creates a temp directory in the dir path provided.
+func (s *ShellTest) Setup(dir string) (*os.File, error) {
+	if err := os.MkdirAll(dir, 0755); err != nil {
+		return nil, err
+	}
+	d, err := ioutil.TempDir(dir, "test")
```

Comment:
Can we use the name of the test? All these random dirs made it hard to know what files belonged to what.

Reply:
yes, I meant to do that.

Reply:
Actually, I meant to use the name in a tempdir, but I like your idea better.  This is my preference to auto-delete...  we clear out the old test and replace it with the new one to keep the clutter down.  It does mean we need to be careful to not give the same test the same name (which I have done when you cut and paste) then run them in parallel.  I'll put a check in for that.